app/build.gradle

```diff
@@ -77,21 +77,36 @@ android {
         }
     }
 
-    File signFile = rootProject.file('signing.properties')
-    if (signFile.exists()) {
-        Properties p = new Properties();
-        p.load(new FileInputStream(signFile))
-
+    if (rootProject.hasProperty("AndroidSigningKeyAlias")
+            && rootProject.hasProperty("AndroidSigningKeyPassword")
+            && rootProject.hasProperty("AndroidSigningStoreFile")
+            && rootProject.hasProperty("AndroidSigningStorePassword")) {
+        println "Using signing properties from gradle properties"
         signingConfigs {
             release {
-                keyAlias p.KeyAlias
-                keyPassword p.KeyPassword
-                storeFile file(p.StoreFile)
-                storePassword p.StorePassword
+                keyAlias rootProject.AndroidSigningKeyAlias
+                keyPassword rootProject.AndroidSigningKeyPassword
+                storeFile file(rootProject.AndroidSigningStoreFile)
+                storePassword rootProject.AndroidSigningStorePassword
             }
+            android.buildTypes.release.signingConfig = signingConfigs.release
         }
-        android.buildTypes.release.signingConfig = signingConfigs.release
+    } else if (System.env.AndroidSigningKeyPassword != null
```

Comment:
Using environment variables allows you to specify the signing properties securely in Jenkins, without them being printed in the Jenkins console log.

Configuration:
<img width="1086" alt="screen shot 2017-05-18 at 11 53 04" src="https://cloud.githubusercontent.com/assets/1731388/26197082/54ca092e-3bc1-11e7-9bf7-dff89dbcdc3a.png">